### PR TITLE
ci: Add Google OAuth provider test

### PR DESCRIPTION
**Pull request by duwenxin99**
1. OAuth provider test must run on Cloud Build to be able to print ID tokens of the GCE VM attached service account.
2. Borrowing gcloud CLI team's public `Client ID` in the Google ID token [documentation](https://cloud.google.com/docs/authentication/token-types#id-contents)

### CODE REVIEW

File: tests/auth_test.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	return string(body), nil
+}
+
+func TestGoogleAuthVerification(t *testing.T) {
+	clientId := "32555940559.apps.googleusercontent.com"
```
Comment (Yuan325) [inline tests/auth_test.go:56]: Is this secure to be included? Or should we put this as a secret?

File: tests/auth_test.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	}
+	for _, tc := range tcs {
+
+		token, err := getGoogleIdToken(clientId)
+		if err != nil {
+			t.Fatalf("ID token generation error: %s", err)
+		}
+		headers := http.Header{}
+		headers.Add("my-google-auth_token", token)
+		_, err = tc.authSource.GetClaimsFromHeader(headers)
```
Comment (Yuan325) [inline tests/auth_test.go]: Is there something that's returned if it's successful? should we test that too?

File: tests/auth_test.go (hunk 1 above)
Comment (kurtisvg) [reply tests/auth_test.go:56]: +1 -- using an env var here should be good?

File: tests/auth_test.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	"testing"
+
+	"github.com/googleapis/genai-toolbox/internal/auth"
+	"github.com/googleapis/genai-toolbox/internal/auth/google"
+)
+
+// Returns an ID token of the service account
+func getGoogleIdToken(audience string) (string, error) {
+	url := "http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/identity?audience=" + audience
+	req, err := http.NewRequest(http.MethodGet, url, nil)
```
Comment (kurtisvg) [inline tests/auth_test.go]: This works with the metadata service, but what about the other ADC? specifically the gcloud login

File: tests/auth_test.go (hunk 1 above)
Comment (duwenxin99) [reply tests/auth_test.go:56]: This is an example client ID provided in the public [documentation](https://cloud.google.com/docs/authentication/token-types#id-contents) so it should be ok.

File: tests/auth_test.go (hunk 2 above)
Comment (duwenxin99) [reply tests/auth_test.go]: Good point. Added.

File: tests/auth_test.go (hunk 3 above)
Comment (duwenxin99) [reply tests/auth_test.go]: added support for local testing using gcloud login credentials

**Pull request merged**